src/sysinfo.cc

```diff
@@ -510,10 +510,13 @@ int GetNumCPUs() {
   int max_id = -1;
   std::ifstream f("/proc/cpuinfo");
   if (!f.is_open()) {
-    std::cerr << "failed to open /proc/cpuinfo\n";
-    return -1;
+    PrintErrorAndDie("Failed to open /proc/cpuinfo");
```

Comment:
White there, could you please cleanup rest of this function, Ctrl+F for `stderr`?